mycroft/skills/core.py

```diff
@@ -231,13 +233,41 @@ def __init__(self, name=None, bus=None):
         self.skill_id = ''  # will be set from the path, so guaranteed unique
         self.voc_match_cache = {}
 
+    @property
+    def enclosure(self):
+        if self._enclosure:
+            return self._enclosure
+        else:
+            LOG.error("ERROR:  Skill not fully initialized.  Move code from " +
+                      " __init__() to initialize() to correct this.")
+            tb = "Traceback:\n"
+            for line in traceback.format_stack()[:-1]:
+                if line.strip():
+                    tb += line
+            LOG.error(tb)
+            raise Exception("Accessed MycroftSkill.enclosure in __init__")
+
+    @property
+    def bus(self):
```

Comment:
Should we add setters for these as well?

```python
@bus.setter
def bus(self, value):
    self._bus = value
```

Reply:
I assume we don't really want people doing a straight "set", they should call MycroftSkill.bind().

Reply:
Fair point. The problem as I see it is when it's used as in padatious_service where ._bus is set and then .bus is used. which looks like a typo when just looking at that file.